firewalld/Dockerfile

```diff
@@ -0,0 +1,46 @@
+# Would it be too large to use fedora as base?
+FROM registry.fedoraproject.org/fedora:latest
+
+ENV VERSION=0 RELEASE=1 ARCH=X86_64
+
+LABEL com.redhat.component="firewalld" \
+      name="$FGC/firewalld" \
+      version="$VERSION" \
+      release="$RELEASE.$DISTTAG" \
+      architecture="$ARCH" \
+      usage="atomic install --system --system-package=no firewalld && systemctl start firewalld" \
+      summary="The firewalld as a system container." \
+      maintainer="Rui xin Bao <rubao@redhat.com>" \
+      atomic.type="system"
+
+RUN dnf install --setopt=tsflags=nodocs -y firewalld && \
+    rpm -V firewalld && \
+    dnf clean all && \
+    mkdir -p /exports/hostfs/var/log/ && \
+    mkdir -p /exports/hostfs/etc/{sysconfig,firewalld} && \
+    mkdir -p /exports/hostfs/usr/share/polkit-1/actions && \
+    mkdir -p /exports/hostfs/etc/dbus-1/system.d && \
+    cp /etc/sysconfig/firewalld /exports/hostfs/etc/sysconfig && \
+    # Copy all the polkit related files into current polkit
+    cp /usr/share/polkit-1/actions/org.fedoraproject.FirewallD1.* /exports/hostfs/usr/share/polkit-1/actions && \
+    # Copy entire firewalld related configuration file into host
+    cp -r /etc/firewalld/* /exports/hostfs/etc/firewalld && \
+    # Copy the dbus configuration file into the host
+    cp /etc/dbus-1/system.d/FirewallD.conf /exports/hostfs/etc/dbus-1/system.d && \
+    # We need /var/log/firewalld for firewalld to write messages
+    touch /exports/hostfs/var/log/firewalld
+
+# Copy the associated binaries to the corresponding location, because firewalld going to use them
+RUN mkdir -p /exports/hostfs/sbin && \
+    mkdir -p /exports/hostfs/usr/sbin && \
+    mkdir -p /exports/hostfs/usr/lib/firewalld && \
+    cp -r /usr/lib/firewalld/* /exports/hostfs/usr/lib/firewalld && \
+    cp /usr/sbin/ebtables /exports/hostfs/usr/sbin/ && \
+    cp /usr/sbin/ipset /exports/hostfs/sbin/
```

Comment:
Yea... I agree it is ugly.. but currently I can't find a neat way to let firewalld find the binaries. During the execution of firewalld, it will call those binaries(ipset and ebatables) for checking. 
https://github.com/firewalld/firewalld/blob/master/src/firewall/config/__init__.py.in#L105
https://github.com/firewalld/firewalld/blob/master/src/firewall/core/fw.py#L76
```
# commands used by backends
COMMANDS = {
    "ipv4":         "/usr/sbin/iptables",
    "ipv4-restore": "/usr/sbin/iptables-restore",
    "ipv6":         "/usr/sbin/ip6tables",
    "ipv6-restore": "/usr/sbin/ip6tables-restore",
    "eb":           "/usr/sbin/ebtables",
    "eb-restore":   "/usr/sbin/ebtables-restore",
    "ipset":        "/usr/sbin/ipset",
    "modinfo":      "/sbin/modinfo",
    "modprobe":     "/sbin/modprobe",
    "rmmod":        "/sbin/rmmod",
}
above is translated commands to loactions when opening firewalld code repo with pycharm
``` 
If I do not have those binaries copied to the host, there will be a WARNING message displayed in the log :(. Is there a good way to let firewalld use those commands while not copying the binaries of the commands to the host? :)

Reply:
how is it using the binaries from the host?  Isn't enough they are in the container?  `/usr` is coming from the container

Reply:
After thinking + reading more documents, there are still a few doubts about the implementation. Any hints would be appreciated =). 
> /usr is coming from container 

1: Then I am a bit confused about how systemd handles paths. I understand that the executable in `/usr/bin/xxx` will all first be resolved as  `/var/lib/containers/atomic/firewalld.0/rootfs/usr/bin` right? But what if `/usr/bin/firewalld` calls other binaries (e.g ipset), during its execution? Does systemd/runc still able to make the path canonical ( relative to the container's rootfs )? 

However, do note this is just a general question about relative path concept, this is not in reply to the answer above.

2: The above confusion also leads to my next question. I found that without copying `/usr/share/polkit-1/actions/xxx` files into the host, the program will fail due to Access denied. But systemd will resolve the path for us right? Why it seems necessary to copy these policy files into host in order to resolve the access denied error ? 

A lot of the decisions( like copying polkit and /usr/lib files) are made just because of try and error, that is why I am a bit tempting to understand the concept behind it. It will be great if there is a document to read through :p.

3: While looking at the files generated for firewalld. I see two important files related to dbus service. One is `/etc/systemd/system/dbus-org.fedoraproject.FirewallD1.service` and another one is
`/etc/dbus-1/system.d/FirewallD.conf`. ( those were used for dbus  I believe)

Then, I think the normal way how firewalld works on the file system (based on inspection and analysis and some guessing) is that it launches dbus --> dbus checks the polkit rules ( uses conf file) and other policies --> dbus launches systemd service through the service file. 

If that is the case, I am a bit unsure about how to link our `service.template` file to this dbus service. As, we had an extra layer of runc in between. I tested a bit with 

```
[Service]
Type=dbus
BusName=org.fedoraproject.FirewallD1
ExecStart=$EXEC_START
ExecStop=$EXEC_STOP
Restart=on-failure
WorkingDirectory=$DESTDIR
RuntimeDirectory=${NAME}
```
will make the service being "active" status. (with type=notify, the service always hang ). But I am a bit unsure about if I did it correctly, and what will the drawbacks might be in the future. WDYT?

4: I am unsure about the relationship between dbus conf and service files, and their relationship when policy kit is involved. Is there any document that is recommended to read in order to shorten this knowledge gap? 

Also, the service file `/etc/systemd/system/dbus-org.fedoraproject.FirewallD1.service` seems to be empty when installing onto a fresh container, probably due to the broken systemd on a container. Any solutions for that, and do we ever want to copy related service files (e.g polkit.service) onto the host? 


**Sorry for these many questions, the knowledge gap is huge for me at the beginning. I was not able to find any meaningful documents to help me make further progress =(. we can have a bluejean chat too if written reply takes too much time. More importantly, thanks for your time =)**

Reply:
once the container runs, the process there has a visibility only on `/var/lib/containers/atomic/firewalld.0/rootfs`, that becomes '/'.  Unless you call executables directly from the systemd service file (which you don't AFAICS), systemd doesn't have any knowledge of how the container looks like and what is inside.

We still need to copy the polkit files to the host and that is correct, since we are using polkit from the host and I think this part works already fine in your current version.  I've played around with it and tried few different things, firewalld was working well.

"Type=dbus" means systemd consider the service started up once it acquires its name from the dbus connection.  You are bind mounting the dbus socket to the container, so firewalld can do that in the same way as it would if running on the host.

The only part I'd like you to investigate is that we can use `/etc/dbus-1/` instead of `/usr/share` for copying the dbus files, as copying files to `/usr/share` goes against the Atomic Host assumption of `/usr` being read only, even if we can circumvent it going into the `/sysroot` path.

And really no worries about the knowledge gap.  You are doing a great job, take your time to understand the (too many) pieces involved.  This is more important than being in a hurry to complete the patch.

Reply:
Thanks for the reply and sorry for the delayed response, was busy with demo and system upgrade(f25 crashed my computer) recently. 

> as copying files to /usr/share goes against the Atomic Host assumption of /usr being read only

I hate to say it, but it seems like both polkit and dbus does not offer a second place for resolving /usr/share related infos( specificlly the policy files) =( .  But if we could do it, please let me know, thanks in advance =).

**TLDR**: Most the below are just proof of why I think finding alternate location might not be doable, so you can skip to the end if you want =). 

``` 
/usr/share is for all read-only architecture independent data files
```
The files that we possibly need for copying to /usr/share are the following ( files not related to /usr directory is ignored) : 
```
[intern3@localhost Downloads]$ rpm -qpl polkit-0.113-15.fc27.x86_64.rpm 
/usr/lib/.build-id
/usr/lib/.build-id/09
/usr/lib/.build-id/09/76bda05cc875e9d0cd823b744eefa0e0168b26
/usr/lib/.build-id/59
/usr/lib/.build-id/59/c8bbcc7fdbbf096fd92183111b07dfd089bc2b
/usr/lib/.build-id/79
/usr/lib/.build-id/79/3aa6d9f308e339c491caf9723ef80228b0f7a6
/usr/lib/.build-id/7f
/usr/lib/.build-id/7f/71dbf9da51a83c4df966234c0814bf108ace44
/usr/lib/.build-id/94
/usr/lib/.build-id/94/e19808053db4a88fe77cff63e97d6ee21ebf3b
/usr/lib/.build-id/c4
/usr/lib/.build-id/c4/3988e95c52a8df70cc770435eb91ab821b8b74
/usr/lib/polkit-1
/usr/lib/polkit-1/polkit-agent-helper-1
/usr/lib/polkit-1/polkitd
/usr/lib/systemd/system/polkit.service
/usr/share/dbus-1/system-services/org.freedesktop.PolicyKit1.service
/usr/share/doc/polkit
/usr/share/doc/polkit/COPYING
/usr/share/doc/polkit/NEWS
/usr/share/doc/polkit/README
/usr/share/locale/cs/LC_MESSAGES/polkit-1.mo
/usr/share/locale/da/LC_MESSAGES/polkit-1.mo
/usr/share/man/man1/pkaction.1.gz
/usr/share/man/man1/pkcheck.1.gz
/usr/share/man/man1/pkexec.1.gz
/usr/share/man/man1/pkttyagent.1.gz
/usr/share/man/man8/polkit.8.gz
/usr/share/man/man8/polkitd.8.gz
/usr/share/polkit-1
/usr/share/polkit-1/actions
/usr/share/polkit-1/actions/org.freedesktop.policykit.policy
/usr/share/polkit-1/rules.d
```
We might also need to include /usr related files from polkit package, I noticed in the normal way of installing firewalld as a package, the systemd unit file comes with it contains `After: polkit` words.

Now, that being said, the following makes me think that dbus/polkit seem not to have methods to handle actions from an alternative place. 

The section below is from dbus man page:  got from man **dbus-daemon**
```
o   <standard_system_servicedirs/>

       <standard_system_servicedirs/> specifies the standard system-wide activation directories that should be searched for service files. As with session services, the first directory listed has highest
       precedence.

       On Unix, the standard session service directories are:

       o   /usr/local/share/dbus-1/system-services: this location is specified by the D-Bus Specification, and is suitable for software installed locally by the system administrator

       o   /usr/share/dbus-1/system-services: this location is specified by the D-Bus Specification, and is suitable for software installed by operating system packages

       o   ${datadir}/dbus-1/system-services for the ${datadir} that was specified when dbus was compiled, typically /usr/share: this location is an extension provided by the reference dbus-daemon
           implementation, and is suitable for software stacks installed alongside dbus-daemon

       o   /lib/dbus-1/system-services: this location is specified by the D-Bus Specification, and was intended for software installed by operating system packages and used during early boot (but it should
           be considered deprecated, because the reference dbus-daemon is not designed to be available during early boot)

```
The section below shows the info from polkit's man page https://cgit.freedesktop.org/polkit/tree/docs/man/polkit.xml#n74 and https://www.freedesktop.org/software/polkit/docs/latest/polkit.8.html
```
 +-------------------+
 |   Authentication  |
 |       Agent       |
 +-------------------+
 | libpolkit-agent-1 |
 +-------------------+
        ^                                  +---------+
        |                                  | Subject |
        +--------------+                   +---------+
                       |                        ^
                       |                        |
User Session           |                        |
=======================|========================|=============
System Context         |                        |
                       |                        |
                       |                    +---+
                       V                    |
                     /------------\         |
                     | System Bus |         |
                     \------------/         |
                       ^        ^           V
                       |        |      +---------------------+
        +--------------+        |      |      Mechanism      |
        |                       |      +---------------------+
        V                       +----> | libpolkit-gobject-1 |
+------------------+                   +---------------------+
|    polkitd(8)    |
+------------------+
| org.freedesktop. |
|    PolicyKit1    |<---------+
+------------------+          |
          ^                   |
          |            +--------------------------------------+
          |            | /usr/share/polkit-1/actions/*.policy |
          |            +--------------------------------------+
          |
   +--------------------------------------+
   | /etc/polkit-1/rules.d/*.rules        |
   | /usr/share/polkit-1/rules.d/*.rules  |
   +--------------------------------------+


and running find on my testing machine outputted the following, all the .policy files are in /usr/share  =(. 
[vagrant@localhost ~]$ sudo find / -name "*.policy"
/usr/share/polkit-1/actions/org.freedesktop.hostname1.policy
/usr/share/polkit-1/actions/org.freedesktop.systemd1.policy
/usr/share/polkit-1/actions/org.freedesktop.policykit.policy
/usr/share/polkit-1/actions/org.freedesktop.login1.policy
/usr/share/polkit-1/actions/org.freedesktop.locale1.policy
/usr/share/polkit-1/actions/org.freedesktop.timedate1.policy
```
===========================Can read from here========================
But one thing to note though, is that both of the code base (dbus and polkit ) seems to mention **datadir** for the policy/service files. I researched it a little bit, and this seems to be defaulted to /usr/share or /usr/local/share directory,  https://www.gnu.org/prep/standards/html_node/Directory-Variables.html

So, may be an alternative would be to configure that(datadir) somehow momentarily? but.. this seems a bit messy. I think we might need to come up with another way of avoiding copying files to /usr. I can't think of a neat way atm, other than copying to the deployment folder on Atomic Host I mentioned earlier. Any other ideas? @giuseppe

**Sorry for the long info** most of it does not require a lot of reading, so I hope that won't take too much of your time, and thanks for your time. =)

Reply:
are you sure that `/usr/local/share` doesn't work already in the same way?  Could you give it a quick try?
`/usr/share/polkit-1/` -> `/usr/local/share/polkit-1/`

Reply:
Hmm, I might have a confusion about concepts, why do the locations in /usr matter? I mean /usr/share/polkit-1 and /usr/local/share/polkit-1 are both under /usr right? I am thinking I am having a misunderstanding of concepts.

Yup, I will try that real quick, for system container (firewalld) on atomic host right?

Reply:
`/usr/local` is a special case, and it is writeable on Atomic Host, so it should work fine

Reply:
Ah, I see, I will try this out, been trying to fight against weird network error lately, sorry if it might take a while, and thanks for your suggestion =D.

Reply:
Sorry for the late response, did not get a chance to work on this until now. In the meantime, may I ask a few extra questions for clarification? Thanks in advance =)

1: how is /usr "read only" enforced on Atomic Host? (ostree managed systems). The root still can modify files in /usr right? In that case, won't it still "betrays" the concept of read only?

**My guess**: is it that at boot time, the image we actually read is not from current /, rather it is from /sysroot/ostree/deploy/fedora-atomic/deploy/xxx?

2: If copying files into /usr is not allowed except /usr/local/share, do we need to take care about files copied into /usr/lib as well? 

**My guess** : /usr/lib is from /lib, but it is still writing to read only /usr, so we need to take care of it? 

3: There are plenty of extra files listed when I did rpm -qpl xxx (package), some of them includes man pages info, or other related/sub binaries. For example, firewalld usually comes with firewall-cmd, do we need to worry about these files too?

**My guess**: those won't be necessary because what we want is only the firewalld's functionality( i.e: to create firewalls), and other files may not be needed to save space / dependency handling

These are doubts of mine, and I make some guesses to them, feel free to correct me if I happen to get it wrong :p. Thanks for your time =D.

Reply:
1. `/usr` is enforced to be read only by a readonly mountpoint.  Only `/usr/local`is writeable: https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html

2. `/usr/lib` is not writeable, what do we still need to copy there?

3. we could copy firewall-cmd to the host (under /usr/local) but for now, let's not copy it and require users to use `atomic run` for launching the command within the container itself.

Reply:
> /usr is enforced to be read only by a readonly mountpoint. Only /usr/local is writeable: https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html

ah I see, yea, it makes sense now, so I assume that any changes made to the existing /usr/local 
will be permanent right? (i.e visible after reboot) 

> /usr/lib is not writeable, what do we still need to copy there?

before, I want to copy those xml files into /usr/lib but later figured that /etc xml seem to work as well,
(files are got from rpm -qlp firewalld package). But I figured, those xml files can be copied to the /etc/firewalld as well, so I think we should be ok =).
```
/usr/lib/firewalld/helpers/Q.931.xml
/usr/lib/firewalld/helpers/RAS.xml
/usr/lib/firewalld/helpers/amanda.xml
/usr/lib/firewalld/helpers/ftp.xml
/usr/lib/firewalld/helpers/h323.xml
/usr/lib/firewalld/helpers/irc.xml
/usr/lib/firewalld/helpers/netbios-ns.xml
/usr/lib/firewalld/helpers/pptp.xml
/usr/lib/firewalld/helpers/sane.xml
/usr/lib/firewalld/helpers/sip.xml
/usr/lib/firewalld/helpers/snmp.xml
/usr/lib/firewalld/helpers/tftp.xml
/usr/lib/firewalld/icmptypes/address-unreachable.xml
/usr/lib/firewalld/icmptypes/bad-header.xml
/usr/lib/firewalld/icmptypes/beyond-scope.xml
/usr/lib/firewalld/icmptypes/communication-prohibited.xml
/usr/lib/firewalld/icmptypes/destination-unreachable.xml
/usr/lib/firewalld/icmptypes/echo-reply.xml
/usr/lib/firewalld/icmptypes/echo-request.xml
/usr/lib/firewalld/icmptypes/failed-policy.xml
/usr/lib/firewalld/icmptypes/fragmentation-needed.xml
/usr/lib/firewalld/icmptypes/host-precedence-violation.xml
/usr/lib/firewalld/icmptypes/host-prohibited.xml
/usr/lib/firewalld/icmptypes/host-redirect.xml
/usr/lib/firewalld/icmptypes/host-unknown.xml
/usr/lib/firewalld/icmptypes/host-unreachable.xml
/usr/lib/firewalld/icmptypes/ip-header-bad.xml
/usr/lib/firewalld/icmptypes/neighbour-advertisement.xml
/usr/lib/firewalld/icmptypes/neighbour-solicitation.xml
/usr/lib/firewalld/icmptypes/network-prohibited.xml
/usr/lib/firewalld/icmptypes/network-redirect.xml
/usr/lib/firewalld/icmptypes/network-unknown.xml
/usr/lib/firewalld/icmptypes/network-unreachable.xml
/usr/lib/firewalld/icmptypes/no-route.xml
/usr/lib/firewalld/icmptypes/packet-too-big.xml
/usr/lib/firewalld/icmptypes/parameter-problem.xml
/usr/lib/firewalld/icmptypes/port-unreachable.xml
/usr/lib/firewalld/icmptypes/precedence-cutoff.xml
/usr/lib/firewalld/icmptypes/protocol-unreachable.xml
/usr/lib/firewalld/icmptypes/redirect.xml
/usr/lib/firewalld/icmptypes/reject-route.xml
/usr/lib/firewalld/icmptypes/required-option-missing.xml
/usr/lib/firewalld/icmptypes/router-advertisement.xml
/usr/lib/firewalld/icmptypes/router-solicitation.xml
/usr/lib/firewalld/icmptypes/source-quench.xml
/usr/lib/firewalld/icmptypes/source-route-failed.xml
/usr/lib/firewalld/icmptypes/time-exceeded.xml
/usr/lib/firewalld/icmptypes/timestamp-reply.xml
/usr/lib/firewalld/icmptypes/timestamp-request.xml
/usr/lib/firewalld/icmptypes/tos-host-redirect.xml
/usr/lib/firewalld/icmptypes/tos-host-unreachable.xml
/usr/lib/firewalld/icmptypes/tos-network-redirect.xml
/usr/lib/firewalld/icmptypes/tos-network-unreachable.xml
/usr/lib/firewalld/icmptypes/ttl-zero-during-reassembly.xml
/usr/lib/firewalld/icmptypes/ttl-zero-during-transit.xml
/usr/lib/firewalld/icmptypes/unknown-header-type.xml
/usr/lib/firewalld/icmptypes/unknown-option.xml
/usr/lib/firewalld/ipsets/README
/usr/lib/firewalld/services/RH-Satellite-6.xml
/usr/lib/firewalld/services/amanda-client.xml
/usr/lib/firewalld/services/amanda-k5-client.xml
/usr/lib/firewalld/services/bacula-client.xml
/usr/lib/firewalld/services/bacula.xml
/usr/lib/firewalld/services/bitcoin-rpc.xml
/usr/lib/firewalld/services/bitcoin-testnet-rpc.xml
/usr/lib/firewalld/services/bitcoin-testnet.xml
/usr/lib/firewalld/services/bitcoin.xml
/usr/lib/firewalld/services/ceph-mon.xml
/usr/lib/firewalld/services/ceph.xml
/usr/lib/firewalld/services/cfengine.xml
/usr/lib/firewalld/services/condor-collector.xml
/usr/lib/firewalld/services/ctdb.xml
/usr/lib/firewalld/services/dhcp.xml
/usr/lib/firewalld/services/dhcpv6-client.xml
/usr/lib/firewalld/services/dhcpv6.xml
/usr/lib/firewalld/services/dns.xml
/usr/lib/firewalld/services/docker-registry.xml
/usr/lib/firewalld/services/dropbox-lansync.xml
/usr/lib/firewalld/services/elasticsearch.xml
/usr/lib/firewalld/services/freeipa-ldap.xml
/usr/lib/firewalld/services/freeipa-ldaps.xml
/usr/lib/firewalld/services/freeipa-replication.xml
/usr/lib/firewalld/services/freeipa-trust.xml
/usr/lib/firewalld/services/ftp.xml
/usr/lib/firewalld/services/ganglia-client.xml
/usr/lib/firewalld/services/ganglia-master.xml
/usr/lib/firewalld/services/high-availability.xml
/usr/lib/firewalld/services/http.xml
/usr/lib/firewalld/services/https.xml
/usr/lib/firewalld/services/imap.xml
/usr/lib/firewalld/services/imaps.xml
/usr/lib/firewalld/services/ipp-client.xml
/usr/lib/firewalld/services/ipp.xml
/usr/lib/firewalld/services/ipsec.xml
/usr/lib/firewalld/services/iscsi-target.xml
/usr/lib/firewalld/services/kadmin.xml
/usr/lib/firewalld/services/kerberos.xml
/usr/lib/firewalld/services/kibana.xml
/usr/lib/firewalld/services/klogin.xml
/usr/lib/firewalld/services/kpasswd.xml
/usr/lib/firewalld/services/kshell.xml
/usr/lib/firewalld/services/ldap.xml
/usr/lib/firewalld/services/ldaps.xml
/usr/lib/firewalld/services/libvirt-tls.xml
/usr/lib/firewalld/services/libvirt.xml
/usr/lib/firewalld/services/managesieve.xml
/usr/lib/firewalld/services/mdns.xml
/usr/lib/firewalld/services/mosh.xml
/usr/lib/firewalld/services/mountd.xml
/usr/lib/firewalld/services/ms-wbt.xml
/usr/lib/firewalld/services/mssql.xml
/usr/lib/firewalld/services/mysql.xml
/usr/lib/firewalld/services/nfs.xml
/usr/lib/firewalld/services/nrpe.xml
/usr/lib/firewalld/services/ntp.xml
/usr/lib/firewalld/services/openvpn.xml
/usr/lib/firewalld/services/ovirt-imageio.xml
/usr/lib/firewalld/services/ovirt-storageconsole.xml
/usr/lib/firewalld/services/ovirt-vmconsole.xml
/usr/lib/firewalld/services/pmcd.xml
/usr/lib/firewalld/services/pmproxy.xml
/usr/lib/firewalld/services/pmwebapi.xml
/usr/lib/firewalld/services/pmwebapis.xml
/usr/lib/firewalld/services/pop3.xml
/usr/lib/firewalld/services/pop3s.xml
/usr/lib/firewalld/services/postgresql.xml
/usr/lib/firewalld/services/privoxy.xml
/usr/lib/firewalld/services/proxy-dhcp.xml
/usr/lib/firewalld/services/ptp.xml
/usr/lib/firewalld/services/pulseaudio.xml
/usr/lib/firewalld/services/puppetmaster.xml
/usr/lib/firewalld/services/quassel.xml
/usr/lib/firewalld/services/radius.xml
/usr/lib/firewalld/services/rpc-bind.xml
/usr/lib/firewalld/services/rsh.xml
/usr/lib/firewalld/services/rsyncd.xml
/usr/lib/firewalld/services/samba-client.xml
/usr/lib/firewalld/services/samba.xml
/usr/lib/firewalld/services/sane.xml
/usr/lib/firewalld/services/sip.xml
/usr/lib/firewalld/services/sips.xml
/usr/lib/firewalld/services/smtp-submission.xml
/usr/lib/firewalld/services/smtp.xml
/usr/lib/firewalld/services/smtps.xml
/usr/lib/firewalld/services/snmp.xml
/usr/lib/firewalld/services/snmptrap.xml
/usr/lib/firewalld/services/spideroak-lansync.xml
/usr/lib/firewalld/services/squid.xml
/usr/lib/firewalld/services/ssh.xml
/usr/lib/firewalld/services/synergy.xml
/usr/lib/firewalld/services/syslog-tls.xml
/usr/lib/firewalld/services/syslog.xml
/usr/lib/firewalld/services/telnet.xml
/usr/lib/firewalld/services/tftp-client.xml
/usr/lib/firewalld/services/tftp.xml
/usr/lib/firewalld/services/tinc.xml
/usr/lib/firewalld/services/tor-socks.xml
/usr/lib/firewalld/services/transmission-client.xml
/usr/lib/firewalld/services/vdsm.xml
/usr/lib/firewalld/services/vnc-server.xml
/usr/lib/firewalld/services/wbem-https.xml
/usr/lib/firewalld/services/xmpp-bosh.xml
/usr/lib/firewalld/services/xmpp-client.xml
/usr/lib/firewalld/services/xmpp-local.xml
/usr/lib/firewalld/services/xmpp-server.xml
/usr/lib/firewalld/xmlschema/check.sh
/usr/lib/firewalld/xmlschema/icmptype.xsd
/usr/lib/firewalld/xmlschema/ipset.xsd
/usr/lib/firewalld/xmlschema/service.xsd
/usr/lib/firewalld/xmlschema/zone.xsd
/usr/lib/firewalld/zones/FedoraServer.xml
/usr/lib/firewalld/zones/FedoraWorkstation.xml
/usr/lib/firewalld/zones/block.xml
/usr/lib/firewalld/zones/dmz.xml
/usr/lib/firewalld/zones/drop.xml
/usr/lib/firewalld/zones/external.xml
/usr/lib/firewalld/zones/home.xml
/usr/lib/firewalld/zones/internal.xml
/usr/lib/firewalld/zones/public.xml
/usr/lib/firewalld/zones/trusted.xml
/usr/lib/firewalld/zones/work.xml
/usr/lib/systemd/system/firewalld.service
```

> we could copy firewall-cmd to the host (under /usr/local) but for now, let's not copy it and require users to use atomic run for launching the command within the container itself.

sure, makes sense =)